db/templates/tbls-specific.jinja.sql

```diff
@@ -114,13 +114,17 @@ ALTER TABLE {{oio_type}}_attr_{{attribut}}_id_seq
 
 CREATE TABLE {{oio_type}}_attr_{{attribut}} (
     id bigint NOT NULL DEFAULT nextval('{{oio_type}}_attr_{{attribut}}_id_seq'::regclass),
-    {%- for field in attribut_fields %} 
-       {{field}} {% if attributter_metadata is defined and attributter_metadata[attribut] is defined and attributter_metadata[attribut][field] is defined and attributter_metadata[attribut][field]['type'] is defined %}
-           {{attributter_metadata[attribut][field]['type']}}
-       {% else %}text{% endif %} null,
-    {%- endfor %} 
-    virkning Virkning not null CHECK( (virkning).TimePeriod IS NOT NULL AND not isempty((virkning).TimePeriod) ),
-    {{oio_type}}_registrering_id bigint not null,
+    {%- for field in attribut_fields %}
+        {{ field }} {% if attributter_metadata is defined and attributter_metadata[attribut] is defined and attributter_metadata[attribut][field] is defined and attributter_metadata[attribut][field]['type'] is defined -%}
+            {{ attributter_metadata[attribut][field]['type'] }}
+        {%- else -%}
+            text
+        {%- endif %} {% if attributter_metadata is defined and attributter_metadata[attribut] is defined and attributter_metadata[attribut][field] is defined and attributter_metadata[attribut][field]['mandatory'] is defined and attributter_metadata[attribut][field]['mandatory'] -%}
```

Comment:
```suggestion
        {%- endif -%}
        {% if attributter_metadata is defined and attributter_metadata[attribut] is defined and attributter_metadata[attribut][field] is defined and attributter_metadata[attribut][field]['mandatory'] is defined and attributter_metadata[attribut][field]['mandatory'] -%}
```

This is about as unreadable earlier line; we should break it up somehow. Perhaps a pre-calculated `dict`? (The above suggestion improves on that a bit, but doesn't really address the issue.)

Reply:
The jinja templates could be cleaner. A lot of these calculations are repeated multiple times and a lot of the if statements in for loops could be removed if we precalculated the right lists only once. This takes time and may be a worthy effort, but has nothing to do with this PR.

Reply:
Fair enough — when writing this, I hadn't noticed that you merely edited these lines 🙂